res/controllers/Numark-Party-Mix.scripts.js

```diff
@@ -0,0 +1,414 @@
+
+    this.connections = new components.ComponentContainer();
+    for (var i = 0; i < 4; i++) {
+        this.connections[i] = new components.SamplerButton({
+            midi: [0x93 + deckNumber, 0x14 + i],
+            number: 1 + i,
+            outConnect: false
+        });
+    }
+};
+NumarkPartyMix.ModeSampler.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkPartyMix.ModeBeatjump = function (deckNumber) {
+    components.ComponentContainer.call(this);
+
+    this.control = NumarkPartyMix.PadModeControls.BEATJUMP;
+
+    const jumps = [
+        "beatjump_1_backward",
+        "beatjump_1_forward",
+        "beatjump_4_backward",
+        "beatjump_4_forward"
+    ];
+
+    this.connections = new components.ComponentContainer();
+    var p = this.connections;
+
+    jumps.forEach(function (jump, i) {
+        p[i] = new components.Button({
+            midi: [0x93 + deckNumber, 0x14 + i],
+            group: "[Channel" + deckNumber + "]",
+            inKey: jump,
+            outKey: jump,
+            outConnect: false
+        });
+    })
+};
+NumarkPartyMix.ModeBeatjump.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkPartyMix.Browse = function () {
+    this.knob = new components.Encoder({
+        input: function (channel, control, value) {
+            var direction;
+            direction = (value > 0x40) ? -1 : 1;
+            engine.setParameter("[Library]", "MoveVertical", direction);
+        }
+    });
+
+    this.knobButton = new components.Button({
+        group: "[Library]",
+        type: components.Button.prototype.types.powerWindow,
+        pressed: false,
+        input: function (channel, control, value, status, group) {
+            if (this.isPress(channel, control, value, status)) {
+                this.inToggle(channel, control, value, status, group);
+                this.isLongPressed = false;
+                this.longPressTimer = engine.beginTimer(this.longPressTimeout, function () {
+                    this.isLongPressed = true;
+                    this.longPressTimer = 0;
+                }.bind(this), true);
+            } else {
+                this.inToggle(channel, control, value, status, group);
+                if (this.longPressTimer !== 0) {
+                    engine.stopTimer(this.longPressTimer);
+                    this.longPressTimer = 0;
+                }
+                this.isLongPressed = false;
+            }
+        },
+        inToggle: function (channel, control, value, status, group) {
+            if (this.isLongPressed) {
+                focused_widget = engine.getParameter("[Library]", "focused_widget");
+                TRACKSTABLE = 3;
+                if (focused_widget === TRACKSTABLE) {
+                    engine.setParameter("[Library]", "focused_widget", 2);
+                } else {
+                    engine.setParameter("[Library]", "focused_widget", 3);
+                }
```

Comment:
Nice to see this being implemented as intended : )

FYI in 2.4/main it also makes (more) sense to focus the searchbar because `[Library], MoveVertical`can scroll the saved queries (new in 2.4).
Did you consider to map `press`+`turn` to moving focus forward/backward?

Reply:
I considered it, but didn't like it. But because I anyway make a 4 deck layout for myself I could make use of this as it let users access more features.
Is there a release schedule for 2.4?

Reply:
Nope, not yet.
Thus, with the longpress+turn instead of focus_widget the mapping could go to 2.3.3 and would be available very soon.
What did you not like about it?

Reply:
> and would be available very soon.

well, as soon as someone finds time to review the mapping. And IIRC there are still a few unfinished mappings / not reviewed completely.

Reply:
changed and rebased to 2.3